include/librealsense2/dds/dds-device-impl.h

```diff
@@ -88,54 +93,89 @@ class dds_device::impl
             throw std::runtime_error( "trying to run() a device that's already running" );
 
         create_notifications_reader();
+        create_control_writer();
         if( ! init() )
             throw std::runtime_error( "failed getting sensors data from device: " + _info.topic_root );
 
         LOG_DEBUG( "Device" << _info.topic_root << " was initialized successfully" );
         _running = true;
     }
 
+    bool write_control_message( void* msg )
+    {
+        assert( _control_writer != nullptr );
```

Comment:
I never understand the value of using assert, it's only relevant for debug right?
Why not checking on release too?

Reply:
_control_writer can be nullptr if there was a call to write_control_message before calling run() (or not handling run failures) 

This kind of initialization logic should be tested at development stage (debug) and the logic does not change when releasing the SW so no point wasting run time on it.